app/experimenter/nimbus-ui/src/index.tsx

```diff
@@ -6,10 +6,18 @@ import React from "react";
 import ReactDOM from "react-dom";
 import App from "./components/App";
 import "./styles/index.scss";
+import { readConfig } from "./lib/config";
 
-ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-  document.getElementById("root"),
-);
+try {
+  const root = document.getElementById("root")!;
+  readConfig(root);
+
+  ReactDOM.render(
+    <React.StrictMode>
+      <App />
+    </React.StrictMode>,
+    root
+  );
```

Comment:
I'm confused about this pattern, how does the `App` read the config?  Is it a hidden shared state inside the config module?

Reply:
When `readConfig` is called above the [config lib](https://github.com/mozilla/experimenter/pull/3670/files#diff-c98cd03bc4fd3949ecfd9580d0d2bc68R19-R24) accesses the encoded data attr value on the mounting element, parsing it and adding it to the config object that can then be imported anywhere in the app.